testsuite/pytests/test_eprop_plasticity.py

```diff
@@ -400,3 +400,133 @@ def generate_superimposed_sines(steps_sequence, periods):
     # Verify results
 
     assert np.allclose(loss, loss_nest_reference, rtol=1e-8)
+
+
+def test_unsupported_surrogate_gradient():
+    """Confirm that selecting a unsupported surrogate gradient raises an error."""
+
+    params_nrn_rec = {
+        "surrogate_gradient_function": "unsupported_surrogate_gradient",
+    }
+
+    with pytest.raises(nest.kernel.NESTError):
+        nrn = nest.Create("eprop_iaf", 1, params_nrn_rec)
+        nest.Simulate(1.0)
+
+
+@pytest.mark.parametrize(
+    "surrogate_gradient_type,surrogate_gradient_reference",
+    [
+        (
+            "piecewise_linear",
+            [
+                0.06995221780833,
+                0.06329538414758,
+                0.05727203196569,
+                0.05182187752951,
+                0.04689037386158,
+            ],
+        ),
+        (
+            "exponential",
+            [
+                0.17906075040984,
+                0.17763607512476,
+                0.17635674663337,
+                0.17520710436201,
+                0.17417332485803,
+            ],
+        ),
+        (
+            "fast_sigmoid_derivative",
+            [
+                0.13052388763403,
+                0.12915920600850,
+                0.12794275148940,
+                0.12685680729831,
+                0.12588607522600,
+            ],
+        ),
+        (
+            "arctan",
+            [
+                0.02631801794450,
+                0.02573637800870,
+                0.02522457486743,
+                0.02477299562220,
+                0.02437357413156,
+            ],
+        ),
+    ],
+)
+def test_eprop_surrogate_gradients(surrogate_gradient_type, surrogate_gradient_reference):
+    """
+    Test correct computation of surrogate gradients by comparing the simulated surrogate gradients with NEST reference
+    surrogate gradients. These reference surrogate gradients were obtained from a simulation with the verified NEST
+    e-prop implementation run with Linux 5.8.7-1-default, Python v3.12.5, Numpy v2.0.1, and NEST@d04fe550d.
+    """  # pylint: disable=line-too-long # noqa: E501
```

Comment:
Here, we do not need to `# pylint: disable=line-too-long #noqa: E501`, since the words are short enough to make line breaks. In the above function we needed to disable it because the the long link to the tutorial file could not be broken over several lines.